src/Processors/Transforms/Streaming/AggregatingHelper.h

```diff
@@ -47,6 +63,16 @@ Chunk mergeAndSpliceAndConvertBucketsToChunk(
 ///         latest update data
 ChunkPair convertToChangelogChunk(AggregatedDataVariants & data, const AggregatingTransformParams & params);
 ChunkPair mergeAndConvertToChangelogChunk(ManyAggregatedDataVariants & data, const AggregatingTransformParams & params);
+
+[[maybe_unused]] ALWAYS_INLINE static bool onlyEmitFinalizedWindows(EmitMode mode) noexcept
+{
+    return mode == EmitMode::Watermark;
+}
+
+[[maybe_unused]] ALWAYS_INLINE static bool onlyEmitUpdates(EmitMode mode) noexcept
+{
+    return static_cast<uint8_t>(mode) & EMIT_UPDATES_MASK;
```

Comment:
Let's revisit this, it appears something wrong here since we are stuffing bit into an enum